packages/api/src/services/emp-stats.ts

```diff
@@ -34,51 +33,76 @@ export default (config: Config, appState: Dependencies) => {
+    if (await hasStatsHistory(empAddress, stat.timestamp)) return stat;
+    // remove id from the latest stats, since this gets overriden with history stats by timestamp
+    const { id, ...historyStat } = stat;
+    return createStatHistory(empAddress, historyStat as EmpHistoryStat);
   }
-
   async function updateHistories(addresses: string[]) {
     return Promise.allSettled(addresses.map(updateHistory));
   }
 
-  async function updateLatestTvl(address: string) {
-    const emp = await queries.getAnyEmp(address);
-    // the full state has collateral decimals, pulled from erc20 state
-    const fullState = await queries.getFullEmpState(emp);
+  // synthetic prices return price in wei, due to the way the bot price feeds work
+  async function getSyntheticPriceFromTable(empAddress: string) {
+    // PriceSample is type [ timestamp:number, price:string]
+    const priceSample: PriceSample = synthPrices.latest[empAddress];
+    assert(uma.utils.exists(priceSample), "No latest synthetic price found for emp: " + empAddress);
 
-    assert(uma.utils.exists(fullState.collateralCurrency), "Emp requires collateralCurrency: " + address);
+    const price = priceSample[1];
+    assert(uma.utils.exists(price), "Invalid latest sythetic price found on emp: " + empAddress);
+    return price;
+  }
 
+  async function getPriceFromTable(empAddress: string, currencyAddress: string) {
     // PriceSample is type [ timestamp:number, price:string]
-    const priceSample: PriceSample = prices[currency].latest[fullState.collateralCurrency];
-    assert(uma.utils.exists(priceSample), "No latest price found for emp: " + address);
-    const [timestamp, price] = priceSample;
-    assert(uma.utils.exists(price), "Invalid latest price found on emp: " + address);
+    const priceSample: PriceSample = prices[currency].latest[currencyAddress];
+    assert(uma.utils.exists(priceSample), "No latest price found for emp: " + empAddress);
+    const [, price] = priceSample;
+    assert(uma.utils.exists(price), "Invalid latest price found on emp: " + empAddress);
+    return price;
+  }
+
+  async function getFullEmpState(empAddress: string) {
+    const emp = await queries.getAnyEmp(empAddress);
+    // the full state has collateral decimals, pulled from erc20 state
+    return queries.getFullEmpState(emp);
+  }
+  async function updateStats(empAddress: string) {
+    const update: empStats.Data = { address: empAddress };
+
+    const emp = await getFullEmpState(empAddress);
+    update.timestamp = nowS();
+
+    if (uma.utils.exists(emp.collateralCurrency)) {
+      update.collateralPrice = await getPriceFromTable(empAddress, emp.collateralCurrency).catch(() => "0");
+      update.tvl = calcTvl(update.collateralPrice, emp).toString();
+    }
+
+    if (uma.utils.exists(emp.tokenCurrency)) {
+      update.syntheticPrice = await getPriceFromTable(empAddress, emp.tokenCurrency).catch(() => "0");
```

Comment:
Instead of providing 0 for prices we don't have, should we have a more well-defined `null` or `undefined` value that can go into our DB? Or did you consider just not updating these if they fail?

Totally not something that's worth a major refactor -- this just might be ambiguous for things where the price actually is zero (expired, worthless option, for instance), but we may never hit that since a lot of exchanges don't allow a price of 0.

Reply:
i really only do it so i can just fall through to the tvl/tvm calculation without an extra check or causing errors. Actually I realized working on future PRs that a lot of this logic needs to be refactored, but that is another issue. 

I'm ok with leaving things null, we will just have to be more careful with dealing with that both on backend and frontend calls.

Reply:
i think the type currently accepts undefined, so i can test setting these to undefined